cpp/src/join/mixed_join_semi.cu

```diff
@@ -197,31 +152,43 @@ std::unique_ptr<rmm::device_uvector<size_type>> mixed_join_semi(
   auto const equality_build_conditional =
     row_comparator_conditional_build.equal_to<false>(build_nulls, compare_nulls);
   double_row_equality equality_build{equality_build_equality, equality_build_conditional};
-  make_pair_function_semi pair_func_build{};
 
-  auto iter = cudf::detail::make_counting_transform_iterator(0, pair_func_build);
+  auto const build_num_rows = compute_hash_table_size(build.num_rows());
+
+  hash_set_type set{build_num_rows,
```

Comment:
In general, avoid using standard keywords as name like this. So instead of `set`, we can call it `row_set` or else.
```suggestion
  hash_set_type row_set{build_num_rows,
```